flow-report/src/common.tsx

```diff
@@ -7,25 +7,62 @@
 import {FunctionComponent} from 'preact';
 
 import {NavigationIcon, SnapshotIcon, TimespanIcon} from './icons';
+import {getScreenDimensions, getScreenshot} from './util';
 
 export const Separator: FunctionComponent = () => {
   return <div className="Separator" role="separator"></div>;
 };
 
+export const FlowStepIcon: FunctionComponent<{mode: LH.Result.GatherMode}> = ({mode}) => {
+  return <>
+    {
+      mode === 'navigation' && <NavigationIcon/>
+    }
+    {
+      mode === 'timespan' && <TimespanIcon/>
+    }
+    {
+      mode === 'snapshot' && <SnapshotIcon/>
+    }
+  </>;
+};
+
 export const FlowSegment: FunctionComponent<{mode?: LH.Result.GatherMode}> = ({mode}) => {
   return (
     <div className="FlowSegment">
       <div className="FlowSegment__top-line"/>
       {
-        mode === 'navigation' && <NavigationIcon/>
-      }
-      {
-        mode === 'timespan' && <TimespanIcon/>
-      }
-      {
-        mode === 'snapshot' && <SnapshotIcon/>
+        mode && <FlowStepIcon mode={mode}/>
       }
       <div className="FlowSegment__bottom-line"/>
     </div>
   );
 };
+
+export const FlowStepThumbnail: FunctionComponent<{
+  reportResult: LH.ReportResult,
+  width?: number,
+  height?: number,
+}> = ({reportResult, width, height}) => {
+  const screenshot = getScreenshot(reportResult);
+
+  // Crop the displayed image to the viewport dimensions.
```

Comment:
Does "viewport" here refer to what was passed in as width/height? If so, could you make that more clear?

Reply:
Viewport refers to the dimensions in the LHR, `width` and `height` are the desired dimensions of the image. Will reword.